src/admin/OpsGuardian.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {ISafe} from "./interfaces/ISafe.sol";
+import {ICreatePool} from "./interfaces/ICreatePool.sol";
+import {IOpsGuardian} from "./interfaces/IOpsGuardian.sol";
+import {IBaseGuardian} from "./interfaces/IBaseGuardian.sol";
+import {IAdapterWiring} from "./interfaces/IAdapterWiring.sol";
+
+import {IAuth} from "../misc/interfaces/IAuth.sol";
+
+import {PoolId} from "../core/types/PoolId.sol";
+import {AssetId} from "../core/types/AssetId.sol";
+import {IAdapter} from "../core/interfaces/IAdapter.sol";
+import {IMultiAdapter} from "../core/interfaces/IMultiAdapter.sol";
+
+contract OpsGuardian is IOpsGuardian {
+    PoolId public constant GLOBAL_POOL = PoolId.wrap(0);
+
+    ISafe public opsSafe;
+    ICreatePool public hub;
+    IMultiAdapter public multiAdapter;
+
+    constructor(ISafe opsSafe_, ICreatePool hub_, IMultiAdapter multiAdapter_) {
+        opsSafe = opsSafe_;
+        hub = hub_;
+        multiAdapter = multiAdapter_;
+    }
+
+    modifier onlySafe() {
+        require(msg.sender == address(opsSafe), NotTheAuthorizedSafe());
+        _;
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Administration
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IBaseGuardian
+    function file(bytes32 what, address data) external onlySafe {
+        if (what == "opsSafe") opsSafe = ISafe(data);
+        else if (what == "hub") hub = ICreatePool(data);
+        else if (what == "multiAdapter") multiAdapter = IMultiAdapter(data);
+        else revert FileUnrecognizedParam();
+        emit File(what, data);
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Adapter Management
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IOpsGuardian
+    function initAdapters(uint16 centrifugeId, IAdapter[] calldata adapters, uint8 threshold, uint8 recoveryIndex)
+        external
+        onlySafe
+    {
+        require(multiAdapter.quorum(centrifugeId, GLOBAL_POOL) == 0, AdaptersAlreadyInitialized());
+        multiAdapter.setAdapters(centrifugeId, GLOBAL_POOL, adapters, threshold, recoveryIndex);
+        IAuth(address(multiAdapter)).deny(address(this));
+    }
+
+    /// @inheritdoc IBaseGuardian
+    function wire(address adapter, uint16 centrifugeId, bytes memory data) external onlySafe {
+        require(!IAdapterWiring(adapter).isWired(centrifugeId), AdapterAlreadyWired());
+        IAdapterWiring(adapter).wire(centrifugeId, data);
+        IAuth(adapter).deny(address(this));
```

Comment:
This should be removed, since the wiring is per network
```suggestion
```

Reply:
I think like William here, the `OpsGuardian` should never call again to `wire` here, isn't?